lib/openstudio-standards/standards/Standards.Space.rb

```diff
@@ -1753,6 +1753,311 @@ def space_design_internal_load(space)
     return load_w
   end
 
+  # Create annual array of occupancy for the space: 1 = occupied, 0 = unoccupied
+  # @author Doug Maddox, PNNL
+  # @ param space object
+  # @return [Double] 8760 array of the occupancy flag
+  def space_occupancy_annual_array(model, space)
+    occ_sch_values = nil
+    ppl_values = Array.new(8760, 0)
+
+    # Need to review all people objects in this space
+    space.spaceType.get.people.each do |people|
+      occ_sch = people.numberofPeopleSchedule
+      if occ_sch.is_initialized
+        occ_sch_obj = occ_sch.get
+        occ_sch_values = get_8760_values_from_schedule(model, occ_sch_obj)
+        # Flag = 1 if any schedule shows occupancy for a given hour
+        if !occ_sch_values.nil?
+          (0..8759).each do |ihr|
+            ppl_values[ihr] = 1 if occ_sch_values[ihr] > 0
+          end
+        else
+          OpenStudio.logFree(OpenStudio::Warn, 'openstudio.standards.Space', "Failed to retrieve people schedule for #{space.name}.  Assuming #{w_per_person}W/person.")
+        end
+      end
+
+    end
+
+    return ppl_values
+  end
+
+  # Determine the design internal load (W) for
+  # this space without space multipliers.
+  # This include People, Lights, Electric Equipment, and Gas Equipment.  
+  # This version accounts for operating schedules
+  # and fraction lost for equipment
+  # @author Doug Maddox, PNNL
+  # @ param space object
+  # @return [Double] 8760 array of the design internal load, in W, for this space
+  def space_internal_load_annual_array(model, space)
+    # For each type of load, first convert schedules to 8760 arrays so coincident load can be determined
+    ppl_values = Array.new(8760, 0)
+    ltg_values = Array.new(8760, 0)
+    load_values = Array.new(8760, 0)
+
+    # DEM: lines to delete
+    #ppl_values = []
+    #ltg_values = []
+    #load_values = []
+
+    #(0..8759).each do |ihr|
+    #  ppl_values[ihr] = 0
+    #  ltg_values[ihr] = 0
+    #  load_values[ihr] = 0
+    #end
```

Comment:
We might want to remove these comments before merging.

Reply:
done